### PR TITLE
 add simple wm8994 codec driver

### PR DESCRIPTION
**Pull request by ChaochaoCui**

### CODE REVIEW

File: boards/arm/stm32f7/stm32f746g-disco/src/stm32_wm8994.c [hunk 1]
```diff
@@ -0,0 +1,231 @@
+/****************************************************************************
+ * boards/arm/stm32f7/stm32f746g-disco/src/stm32_wm8994.c
+ *
+ *   Copyright (C) 2019 Gregory Nutt. All rights reserved.
```
Comment (xiaoxiang781216) [inline boards/arm/stm32f7/stm32f746g-disco/src/stm32_wm8994.c:4]: need change apache license

File: include/nuttx/audio/wm8994.h [hunk 2]
```diff
@@ -0,0 +1,277 @@
+/****************************************************************************
+ * include/nuttx/audio/wm8994.h
+ *
+ *   Copyright (C) 2019 Gregory Nutt. All rights reserved.
```
Comment (xiaoxiang781216) [inline include/nuttx/audio/wm8994.h:4]: change to apache license too

File: drivers/audio/wm8994.c [hunk 3]
```diff
@@ -0,0 +1,2033 @@
+/***********************************************************************************
+ * drivers/audio/wm8994.c
+ *
+ * Audio device driver for Wolfson Microelectronics WM8994 Audio codec.
+ *
+ *   Copyright (C) 2014, 2016-2019 Gregory Nutt. All rights reserved.
```
Comment (xiaoxiang781216) [inline drivers/audio/wm8994.c:6]: change to apache license

File: drivers/audio/wm8994_debug.c [hunk 4]
```diff
@@ -0,0 +1,565 @@
+/*****************************************************************************************
+ * drivers/audio/wm8994_debug.c
+ *
+ * Audio device driver for Wolfson Microelectronics WM8904 Audio codec.
+ *
+ *   Copyright (C) 2019 Gregory Nutt. All rights reserved.
```
Comment (xiaoxiang781216) [inline drivers/audio/wm8994_debug.c:6]: change to apache license

File: drivers/audio/wm8994.h [hunk 5]
```diff
@@ -0,0 +1,1658 @@
+/***************************************************************************************************
+ * drivers/audio/wm8994.h
+ *
+ *   Copyright (C) 2019 Gregory Nutt. All rights reserved.
```
Comment (xiaoxiang781216) [inline drivers/audio/wm8994.h:4]: change to apache license

**Comment by xiaoxiang781216** [comment]
@ChaochaoCui could you fix the warning:
Warning: /home/runner/work/incubator-nuttx/incubator-nuttx/nuttx/drivers/audio/wm8994.h:145:62: warning: Wrong column position of comment right of code

File: drivers/audio/wm8994.h (hunk 5 above)
Comment (ChaochaoCui) [reply drivers/audio/wm8994.h:4]: OK

**Comment by xiaoxiang781216** [comment]
@ChaochaoCui don't need close the PR. you can update the PR locally by:
git push -f origin stm32f746g-disco_audio

**Pull request closed without merging**